File: photomap-core-api/photo-api/src/main/java/ua/in/photomap/photoapi/websocket/NotificationSender.java
package ua.in.photomap.photoapi.websocket;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import ua.in.photomap.common.photo.model.dto.websocket.UserPhotoNotificationDTO;

import java.util.Map;

@Service
@RequiredArgsConstructor
@Slf4j
public class NotificationSender {

    private final RestTemplate restTemplate;

    @Value("${photomap.notification-api.base-url}")
    private String notificationApiBaseUrl;

    public void sendUserPhotoNotification(UserPhotoNotificationDTO message) {
        ResponseEntity<String> response = restTemplate.postForEntity(notificationApiBaseUrl + "/ws/user/photo", message, String.class);
        log.info("Notification sent. Response: " + response);
    }
}

File: photomap-core-api/photo-api-python/websocket/notification_sender.py
import logging
import requests
from dataclasses import asdict
from dto.user_photo_notification_dto import UserPhotoNotificationDTO

logger = logging.getLogger(__name__)

class NotificationSender:
    
    def __init__(self, notification_api_base_url: str, timeout: int):
        self.notification_api_base_url = notification_api_base_url
        self.timeout = timeout
    
    def send_user_photo_notification(self, message: UserPhotoNotificationDTO):
        endpoint = f"{self.notification_api_base_url}/ws/user/photo"
        
        response = requests.post(
            endpoint,
            json=asdict(message),
            timeout=self.timeout
        )
        
        logger.info(f"Notification sent. Response: {response.status_code} - {response.text}")
        response.raise_for_status() 